timely_beliefs/beliefs/utils.py

```diff
@@ -37,11 +56,25 @@ def replace_multi_index_level(df: "classes.BeliefsDataFrame", level: str, index:
     df = df.reset_index()
 
     # Replace desired index level (as a column)
-    if isinstance(index, DatetimeIndex):
+    if isinstance(index, pd.DatetimeIndex):
         df[level] = index.to_series(keep_tz=True, name="").reset_index()[index.name]
     else:
         df[level] = index.to_series(name="").reset_index()[index.name]
     df = df.rename({level: index.name}, axis="columns")
 
     # Convert columns to index levels (only columns that represent index levels)
     return df.set_index(indices)
+
+
+def load_time_series(
+    series: pd.Series, sensor: Sensor, source: BeliefSource, horizon: timedelta
```

Comment:
I suggest renaming `horizon` to `belief_horizon` and possibly also `series` to `event_value_series`. Would also be nice to be able to specify `belief_time` rather than `belief_horizon`.